### PR TITLE
Depend on haskell-mode

### PR DESCRIPTION
**Pull request by glyn**
**Note to reviewers: this PR is untested.**

Fixes https://github.com/emacs-lsp/lsp-haskell/issues/163

### CODE REVIEW

File: lsp-haskell.el
Comment (glyn) [inline lsp-haskell.el:4]: 16.1 appears to be current at the time of writing. I'm not really sure what to put here.

File: Eask
Comment (glyn) [inline Eask:2]: I considered 1.0.1, but requiring haskell-mode feels like a significant change so 1.1 or 2.0 is probably more appropriate, depending on the versioning policy in use.

**Comment by glyn** [comment]
CI failed:
2022-12-30-184407_905x749_scrot

There seems to be a duplicate dependency on `haskell-mode`. Not sure what is causing this.

File: Eask
Comment (jcs090218) [inline Eask:17]: There is another `(depends-on "haskell-mode")` on line 25. You should the one in `(development  ..` scope.

**Review by jcs090218 — COMMENTED** [review]
That should fix the CI, see review.

File: Eask
Comment (glyn) [reply Eask:17]: Make sense, thanks. Done.

**Comment by glyn** [comment]
@jcs090218 Please would you approve the workflow so we can see if CI is fixed? Also, kindly PTAL at my other comments in this PR.

File: Eask
Comment (jcs090218) [reply Eask:2]: Make sense, but let's wait what other people think! Thanks!

**Comment by jcs090218** [comment]
> Please would you approve the workflow so we can see if CI is fixed?

I don't have permission to this repo; let me ping repo owners. They should be able to help us!

cc @yyoncho @alanz

**Review by jcs090218 — APPROVED** [review]
As I mentioned above, I don't have permission, but the code LGTM! :)

Thanks for updating the README as well! ❤️ 👍

**Review by michaelpj — APPROVED** [review]
Looks great, thanks!

**Comment by glyn** [comment]
> Looks great, thanks!

On the plus side, you're welcome. On the other hand, eek, let's hope it works!

**Pull request merged**
